### PR TITLE
feat(e2e): Add comprehensive signal-decision engine test coverage

### LINKED ISSUE #692: [E2E] Add Signal-Decision Engine Test Coverage for New Plugin Architecture
## Description

Add E2E test coverage for the signal-decision engine and plugin architecture introduced in #681. This complements #661 by testing the new decision-based routing mechanics.

## Background

PR #681 introduced a breaking architectural change from category-based routing to a signal-decision engine with modular plugins. 

**Key concepts needing tests:**
- **Decisions**: Priority-based routing with rules
- **Plugins**: Modular chains (PII → cache → system prompts)
- **Signals**: Input conditions (domain, keyword, entropy)
- **Strategy**: Decision selection (priority vs confidence)

### 1. Decision Priority & Strategy
- Multiple decisions matching → highest priority wins
- Confidence-based strategy vs priority-based
- Fallback to default when no match

### 2. Plugin Chain Execution
- Plugins execute in order
- PII plugin blocks entire chain when triggered
- Cache plugin works after PII passes
- System prompt applied after cache miss

### 3. Rule Condition Logic
- AND operator (all conditions must match)
- OR operator (any condition matches)
- Keyword matching (case-sensitive/insensitive)
- Domain + keyword combination

### 4. Plugin Configuration
- PII allowlist (`pii_types_allowed: ["PERSON"]`)
- Per-decision cache thresholds
- Plugin-specific configurations

### 5. Edge Cases
- No matching decision → fallback behavior
- Invalid configuration → validation errors
- Plugin conflicts (PII blocks + cache enabled)

## Example Test

```
func testDecisionPriority(ctx context.Context, client *kubernetes.Clientset, opts testcases.TestCaseOptions) error {
    // Query matches both "thinking" (priority 15) and "business" (priority 10)
    query := "Think carefully about this business decision"
    response := sendChatRequest(opts.EnvoyService, query)
    
    // Verify highest priority won
    if response.Headers["X-VSR-Selected-Decision"] != "thinking_decision" {
        return fmt.Errorf("expected thinking_decision, got %s", response.Headers["X-VSR-Selected-Decision"])
    }
    return nil
}
```

## Implementation Plan

**Phase 1: Core Logic**
- Decision priority/strategy selection
- Rule conditions (AND/OR)
- Fallback behavior

**Phase 2: Plugin Chains**
- Execution order
- Blocking behavior
- Configuration variations

**Phase 3: Integration**
- Multi-signal routing
- Plugin interactions
- Error handling

## Acceptance Criteria

- [ ] Decision priority and strategy selection tested
- [ ] Plugin chain execution order verified
- [ ] Rule condition logic (AND/OR) covered
- [ ] Plugin configuration variations tested
- [ ] Documentation updated in `e2e/README.md`

## Related Issues

- #681 - Signal-Decision architecture PR (breaking changes)

### PR DESCRIPTION
**Pull request by yehuditkerido**
Implements 6 new E2E test suites to validate the signal-decision
engine :

New Test Files:
- decision_priority.go: Priority-based routing (4 cases)
- plugin_chain_execution.go: Plugin ordering & blocking (4 cases)
- rule_condition_logic.go: AND/OR operators (6 cases)
- decision_fallback.go: Fallback behavior (5 cases)
- keyword_routing.go: Keyword matching (6 cases)
- plugin_config_variations.go: Plugin configs (6 cases)

Key Fixes:
- Fix keyword_rules field: 'category' → 'name'
- Fix rule condition field: 'rule_name' → 'name'
- Add keyword conditions to thinking_decision
- Standardize fallback decision name: 'general_decision' → 'other_decision'

Additional Changes:
- Add signal-decision tests to all 3 profiles (ai-gateway, aibrix, dynamic-config)
- Fix markdown formatting in e2e/README.md and website/docs/api/crd-reference.md

Resolve #692

### CODE REVIEW

**Review by Xunzhuo — COMMENTED** [review]
hopefully you can catch some bugs after adding the testcases, happy hacking! thanks

**Comment by Xunzhuo** [comment]
We need to add this to dynamic config profile as well, plz check the relevant files

File: e2e/profiles/dynamic-config/crds/intelligentroute.yaml [hunk 1]
```diff
@@ -32,10 +32,43 @@ spec:
...
+
   decisions:
+    - name: "thinking_decision"
+      priority: 15
+      description: "Queries requiring careful thought or urgent attention"
+      signals:
+        operator: "OR"
+        conditions:
+          - type: "domain"
+            name: "thinking"
```
Comment (Xunzhuo) [inline e2e/profiles/dynamic-config/crds/intelligentroute.yaml]: we dont have a domain called thinking

**Comment by yehuditkerido** [comment]
Hi @Xunzhuo.
I've added 6 new signal-decision engine tests across all 3 profiles. While implementing them, I discovered some (probably) backend issues:
### Test Accuracy:
    • ✅ decision-priority-selection: 100% (working perfectly)
    • ⚠️ plugin-chain-execution: 75% (US_SSN detection works, EMAIL_ADDRESS doesn't)
    • 🔴 keyword-routing: 17% (only "ASAP" keyword works)
    • 🔴 rule-condition-logic: 33% (keywords work sporadically)
    • 🔴 decision-fallback-behavior: 40% (classifier unreliable)
    • ⚠️ plugin-config-variations: 50% (same classifier issues)
### Quick Summary of Issues:
    1. PII: EMAIL_ADDRESS not blocking (returns 200 instead of 403), but US_SSN works fine
    2. Keywords: Only "ASAP" matches consistently, others ("urgent", "think", "careful") fail
    3. Domain Classifier: Returns empty/wrong classifications for obvious queries
I hope I have verified correctly the test implementations (proper headers, config, model parameter). I suggest the issues are backend bugs, but feel free to double check this conclusion. Should I open separate issues for these backend bugs, or would you prefer to fix them before merging? Tests currently "pass" in CI (0% threshold) so they won't block anything. Anyway I will be happy to take part in those future investigations.

**Review by Xunzhuo — APPROVED** [review]

**Comment by Xunzhuo** [comment]
Thanks!

**Comment by Xunzhuo** [comment]
@yehudit1987 would you like to try adding test cases around embedding signals?

**Comment by Xunzhuo** [comment]
And plz create issues for your findings, if u can dig more that will be helpful

**Comment by yehuditkerido** [comment]
> And plz create issues for your findings, if u can dig more that will be helpful

I will, please leave it for me.  Will create those next week and ask for your approval.
